renaming TestDdpCommHook class so it doesn't get picked up as a test by pytest

bdhirsh opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #44906 adding a test for ddp save()/load()
* **#44905 renaming TestDdpCommHook class so it doesn't get picked up as a test by pytest**

Differential Revision: [D23825308](https://our.internmc.facebook.com/intern/diff/D23825308)

bdhirsh committed:
renaming TestDdpCommHook class so it doesn't get picked up as a test by pytest

[ghstack-poisoned]

mrshenli commented on test/distributed/test_c10d.py:
```diff
@@ -1936,7 +1936,7 @@ def forward(self, x):
 
 
-class TestDdpCommHook(nn.Module):
+class DdpComHook(nn.Module):
```
com -> comm

Shall we rename this to `ModuleForDdpCommHook` as this is a module instead of a hook.

mrshenli reviewed (approved):
LGTM!

cc @pritamdamania87, for context, Brian discovered that when using the name `TestDdpCommHook`, pytest might mis-identify this as a test class, and hence we removed the `Test` prefix. Please let us know if this rename is not OK. Thanks!

pritamdamania87 commented:
> cc @pritamdamania87, for context, Brian discovered that when using the name TestDdpCommHook, pytest might mis-identify this as a test class, and hence we removed the Test prefix. Please let us know if this rename is not OK. Thanks!

I think the rename makes sense, thanks for fixing this :)

bdhirsh committed:
Update on "renaming TestDdpCommHook class so it doesn't get picked up as a test by pytest"




[ghstack-poisoned]

bdhirsh committed:
Update on "renaming TestDdpCommHook class so it doesn't get picked up as a test by pytest"


Differential Revision: [D23825308](https://our.internmc.facebook.com/intern/diff/D23825308)

[ghstack-poisoned]
